sys/include/net/sock/ip.h

```diff
@@ -389,10 +390,9 @@ int sock_ip_get_remote(sock_ip_t *sock, sock_ip_ep_t *ep);
  * @param[out] data     Pointer where the received data should be stored.
  * @param[in] max_len   Maximum space available at @p data.
  * @param[in] timeout   Timeout for receive in microseconds.
- *                      This value can be ignored (no timeout) if the
- *                      @ref sys_xtimer module is not present or the
- *                      implementation does not support timeouts on its own.
- *                      May be 0 for no timeout.
+ *                      If 0 and no data is available, return immediately.
```

Comment:
> Yes, better

Then this is null and void:

> I wouldn't know how to implement such a case with stacks where 0 means no timeout... I would set it to one, maybe...

Then please add the return value `-EWOULDBLOCK` or `-EAGAIN`.

> I'll change if you approve the change in general.

I still see some implementation confusions, but I guess that is the reviewers job to spot these ;-)